pennylane/devices/default_qubit.py

```diff
@@ -621,6 +621,24 @@ def vn_entropy(self, wires, log_base):
         state = self._flatten(self._pre_rotated_state)
         return qml.math.to_vn_entropy(state, indices=wires, c_dtype=self.C_DTYPE, base=log_base)
 
+    def mutual_info(self, wires0, wires1, log_base):
+        """Returns the mutual information between the subsystems.
+
+        Args:
+            wires0 (Wires): wires of the first subsystem.
+            wires1 (Wires): wires of the second subsystem
+            log_base (float): base to use in the logarithm.
+
+        Returns:
+            float: the mutual information
+        """
+        wires0 = wires0.tolist()
+        wires1 = wires1.tolist()
+        state = self._flatten(self._pre_rotated_state)
+        return qml.math.to_mutual_info(
+            state, indices0=wires0, indices1=wires1, c_dtype=self.C_DTYPE, base=log_base
+        )
+
```

Comment:
I think it would be great to move this definition into `QubitDevice` such that it uses the `access_state` method. That way, we provide a single definition for plugin state vector simulators too. Error handling could be done e.g., like:
```py
            try:
                state = self.access_state()
            except qml.QuantumFunctionError as e:
                raise NotImplementedError(
                    f"Cannot compute the mutual information of device {self.name} that is not capable of returning the state."
                ) from e
```
Note: the snippet is untested and the indentation might be off.

Reply:
@eddddddy I am changing it in the entropy PR first, let's see if it works well